src/realm/util/config.h.in

```diff
@@ -13,6 +13,8 @@
 #cmakedefine01 REALM_ENABLE_MEMDEBUG
 #cmakedefine01 REALM_VALGRIND
 #cmakedefine01 REALM_METRICS
+#cmakedefine01 REALM_ASAN
+#cmakedefine01 REALM_TSAN
```

Comment:
Where are these coupled to the respective compilation modes?

Reply:
[CMakeLists.txt:130](https://github.com/realm/realm-core/blob/485aaaef402b669d6ec24f632c1a44afd26ec390/CMakeLists.txt#L131) generates the config.h file. The syntax here is shorthand for substituting the value of the cmake variable to a C define of the same name. The definitions for the build modes of REALM_ASAN and REALM_TSAN are in the [tools/cmake/SpecialtyBuilds.cmake](https://github.com/realm/realm-core/blob/485aaaef402b669d6ec24f632c1a44afd26ec390/tools/cmake/SpecialtyBuilds.cmake#L40) file.